internal/datanode/data_node.go

```diff
@@ -314,6 +316,13 @@ func (node *DataNode) handleWatchInfo(e *event, key string, data []byte) {
 			return
 		}
 
+		if watchInfo.State != datapb.ChannelWatchState_ToWatch && // start watch
+			watchInfo.State != datapb.ChannelWatchState_ToRelease && // start release
+			watchInfo.State != datapb.ChannelWatchState_Uncomplete { // legacy state, equal to ToWatch
+			log.Warn("watch channel state in result state", zap.String("state", watchInfo.State.String()))
```

Comment:
How about something like "Data node is handling a PUT event, while the channel watch info was in wrong state"